tasks/adventure_solutions/task_7_while.py

```diff
@@ -0,0 +1,81 @@
+import random
+import time
+
+def wrong_input(valid):
+    print(f"Sorry but the only valid options are: {valid}.")
+
+def dead(death_message):
+    print(f"Sorry {name}, {death_message}.")
+    print("You're falling for...")
+
+    for number in range(1,11):
+        print(number)
+        time.sleep(1)
+
+    print("meters!")
+    print("You are dead!")
+
+friends = ['Emily', 'Aurora', 'Jack', 'Daniel', 'Meghan']
+
+door_greetings = {'1': "Welcome to paradise!",
+                  '2': "Welcome to the dormitory!"}
+
+print()
+name = input("What is your name? ")
+print(f"Welcome to the dungeon, {name}!")
+
+print("Do you go through door 1 or door 2?")
+door = input("> ")
+
+while door != "1" and door != "2":
```

Comment:
To avoid having the 1 or 2 question twice and the door input twice and then the `else` in the end with wrong_input that is in theory dead code and it should never get there, I would have structured it a bit different, adding the initial question and first door input together with the door handling inside the while block.